tree/ntuple/v7/test/ntuple_field_iterator.cxx

```diff
@@ -23,16 +23,16 @@ TEST(RNTupleDescriptor, FieldIterator)
 
     printf("-- printing top-level fields\n");
     const auto& ntuple_desc = ntuple.GetDescriptor();
-    for (auto f: ntuple_desc.GetTopLevelFields()) {
-       printf("top level field id: %lu\n", f);
-       printf("\tname: %s\n", ntuple_desc.GetFieldDescriptor(f).GetFieldName().c_str());
+    for (auto& f: ntuple_desc.GetTopLevelFields()) {
+       printf("top level field id: %lu\n", f.GetId());
+       printf("\tname: %s\n", f.GetFieldName().c_str());
```

Comment:
I think iterating over field descriptor reference is arguably a nicer interface, but there are some downsides. 

- You need `auto& f` because we can't make descriptor copies, so `auto f` won't compile. 
```
/home/max/projects/root/tree/ntuple/v7/test/ntuple_field_iterator.cxx:26:48: 
error: use of deleted function ‘ROOT::Experimental::RFieldDescriptor::RFieldDescriptor(const ROOT::Experimental::RFieldDescriptor&)’
     for (auto f: ntuple_desc.GetTopLevelFields()) {
```
- The implementation is also more convoluted, and maybe there are dangling pointer concerns if the `NTupleDescriptor` goes out of scope.